http_parser.c

```diff
@@ -472,6 +483,109 @@ static struct {
 
 int http_message_needs_eof(const http_parser *parser);
 
+static const char* find_crlf(const char* p, const char* data, size_t len);
```

Comment:
I have a proposal. What do you think about moving this into a separate header file? This way we could write extensive tests for this function.

Reply:
so do you mean that we create another header file or do you just mean moving it to http_parser.h?

Or could we perhaps have a non-static function that is conditionally included based on some flags that calls this static function? if not this kinda gets exposed unnecessarily.

Reply:
I'd just put it into internal header file, so that it could be included and covered by tests.

Reply:
ok, got it, let me do that in a while.